### ISSUE TITLE
Function size

### ISSUE THREAD

**Issue by wallcrawler** [FAC-GM/app#17]
(This)[https://github.com/FAC-GM/app/blob/master/lib/home.js#L61] handler is huge. The front end JS is very readable due to the small functions, whereas this one is very difficult to grasp as a reader.

**Comment by nelsonic** [comment]
@wallcrawler agree, the `handler` *looks* big at [***58 Lines***](https://github.com/FAC-GM/app/blob/22f104d0a5c45d116b015f4ca67369e141239757/lib/home.js#L61-L119) *however* with some tidying it goes down to **45** (*which is not* ***ideal*** *its quite clear what its meant to do...*) and with a few clarifying comments its not *that* complex.
Provided there aren't *too* many branches its actually quite readable.

**Comment by iteles** [comment]
Action: Review handler to ensure it is well commented and clear in its intent.
